### PR TITLE
Text tab support and 'my tabs'

### PR DESCRIPTION
**Pull request by steveandroulakis**
Adds the 'Text' option to the UI
- Copies tabs from Ultimate Guitar and writes them to text files
- Works for Chords, Tabs, Bass and Ukulele type tabs

Adds the ability to download 'My Tabs' (your saved tabs)
- Works for all tab types
- Inserts into the current artist/tab file structure

Code could be cleaned up a bit, but in my testing it works!

### CODE REVIEW

**Comment by jabbey1** [comment]
Hey, thanks for contributing- I'm interested in these features. Didn't even know there was a 'my tabs' section. It's not currently working on my end though, I think they may have added another popup after logging in that's interfering, but strangely it's only happening with your updates. I'll investigate.

File: src/UGDownloader/CTKGUI.py [hunk 1]
```diff
@@ -177,20 +177,24 @@ def __init__(self, ):
...
+        self.download_button.grid(row=10, column=4, pady=(10, 20), padx=10)
         self.progress_bar.set(0)
 
         """GUI arrangement over"""
 
         # set default values
         self.browser_button.set('Chrome')
         self.headless_checkbox.select()
-        self.filetype_drop_down.set('Guitar Pro')
+        self.filetype_drop_down.set('Text')
```
Comment (jabbey1) [inline src/UGDownloader/CTKGUI.py:197]:
```suggestion
        self.filetype_drop_down.set('Guitar Pro')
```

**Comment by steveandroulakis** [comment]
Nice, solved the bug?

**Comment by jabbey1** [comment]
Wasn't even a bug really, just missed it the first time I looked and didn't get back to it for awhile.
There's the section you commented out for clicking on a popup after you login- understandable if you don't get the popup (it's a membership promotion), would have been an annoying wait there. I did have the wait time as 20 seconds which is a little ridiculous so I shortened it to 5, maybe that's still too long but it does have a little delay before appearing.

**Comment by Rexadev** [comment]
Please add ability to resize the window. I have to hide taskbar to be able to use the program

**Comment by Rexadev** [comment]
Option to individually search each (Chords, Tabs, Bass, Ukulele) would be great because it's gets stuck after search for Chords and tabs.

**Comment by Rexadev** [comment]
Why do you call Chords chord tabs ?

**Comment by Rexadev** [comment]
@jabbey1 not drums ?

**Comment by Rexadev** [comment]
Is it possible to save Chords to chordpro file ?

**Comment by steveandroulakis** [comment]
Thanks for the feedback! Feel free to make a PR for any of these. I'll take a look sometime (ETA in the next month).

**Comment by Rexadev** [comment]
@jabbey1 How text UGDownloader says `File already exists. Skipping.` but for Power/Pro Tabs it redownloads
https://github.com/jabbey1/UGDownloader/issues/14

**Comment by jabbey1** [comment]
Text tabs are newly created .txt files. Other files are downloaded through your browser like you would download a file normally while browsing.

**Comment by Rexadev** [comment]
# [Bug Report] Error Download Taylor Swift Text Tabs

```
Checking for updated .exe release.
No new release found. Current version: 2.5
Currently, no cookies pop-up removing add-on is included for Firefox, please try Chrome instead if you are having cookies pop-up problems.

Starting Firefox. Downloading Geckodriver.

Starting download...
Logging in...
Logged in
Grabbing urls of requested files.

Reading page 1 for Chords tabs
Reading page 2 for Chords tabs
Reading page 3 for Chords tabs
Reading page 4 for Chords tabs
Reading page 5 for Chords tabs
Reading page 6 for Chords tabs
Reading page 7 for Chords tabs
Reading page 8 for Chords tabs
Reading page 9 for Chords tabs
Reading page 10 for Chords tabs
Reading page 11 for Chords tabs
Found 1099 Chords files

Reading page 1 for Tab tabs
Reading page 2 for Tab tabs
Reading page 3 for Tab tabs
Reading page 4 for Tab tabs
Reading page 5 for Tab tabs
Found 422 Tab files

Reading page 1 for Bass tabs
Found 98 Bass files

There are no available Ukulele tabs for this artist.
Attempting 1619 downloads.

Taylor Swift - 1989 Album - Chords
File already exists. Skipping.


Taylor Swift - 1989 Taylors Version Album* - Chords
File already exists. Skipping.


Taylor Swift - 1989 Taylors Version Album (ver 2)* - Chords
File already exists. Skipping.


Taylor Swift - Beautiful Eyes Album - Chords
File already exists. Skipping.


Taylor Swift - Evermore Album* - Chords
File already exists. Skipping.


Taylor Swift - Fearless Album - Chords
File already exists. Skipping.


Taylor Swift - Fearless Album (ver 2) - Chords
File already exists. Skipping.


Taylor Swift - Fearless Taylors Version Album* - Chords
File already exists. Skipping.


Taylor Swift - Folklore Album - Chords
File already exists. Skipping.


Taylor Swift - Lover Album* - Chords
File already exists. Skipping.


Taylor Swift - Midnights Album - Chords
File already exists. Skipping.


Taylor Swift - Red Album* - Chords
File already exists. Skipping.


Taylor Swift - Red Album (ver 2)* - Chords
File already exists. Skipping.


Taylor Swift - Red Album (ver 3)* - Chords
File already exists. Skipping.


Taylor Swift - Red Album Acoustic* - Chords
File already exists. Skipping.


Taylor Swift - Red Taylors Version Album* - Chords
File already exists. Skipping.


Taylor Swift - Reputation Album - Chords
File already exists. Skipping.


Taylor Swift - Speak Now Album - Chords
File already exists. Skipping.


Taylor Swift - Speak Now Taylors Version Album* - Chords
File already exists. Skipping.


Taylor Swift - Taylor Swift Album* - Chords
File already exists. Skipping.


Taylor Swift - Taylor Swift Album (ver 2)* - Chords
File already exists. Skipping.


Taylor Swift - The Tortured Poets Department Album* - Chords
File already exists. Skipping.


Taylor Swift - Birch - Chords
File already exists. Skipping.


Taylor Swift - Birch (ver 2) - Chords
File already exists. Skipping.


Taylor Swift - Both Of Us - Chords
Downloading tab text @ https://tabs.ultimate-guitar.com/tab/b-o-b/both-of-us-chords-1147117

```

**Comment by Rexadev** [comment]
3. [Bug]Not Searching for ukulele it says `There are no available Ukulele tabs for this artist.` but there are ukulele tabs here https://www.ultimate-guitar.com/search.php?title=green+day&page=1&type=800
```
Starting Firefox. Downloading Geckodriver.

Reading page 1 for Chords tabs
Reading page 2 for Chords tabs
Reading page 3 for Chords tabs
Reading page 4 for Chords tabs
Reading page 5 for Chords tabs
Reading page 6 for Chords tabs
Reading page 7 for Chords tabs
Found 648 Chords files

Reading page 1 for Tab tabs
Reading page 2 for Tab tabs
Reading page 3 for Tab tabs
Reading page 4 for Tab tabs
Reading page 5 for Tab tabs
Reading page 6 for Tab tabs
Reading page 7 for Tab tabs
Reading page 8 for Tab tabs
Reading page 9 for Tab tabs
Reading page 10 for Tab tabs
Found 977 Tab files

Reading page 1 for Bass tabs
Reading page 2 for Bass tabs
Reading page 3 for Bass tabs
Reading page 4 for Bass tabs
Reading page 5 for Bass tabs
Reading page 6 for Bass tabs
Found 549 Bass files

There are no available Ukulele tabs for this artist.

Your Green Day folder has 845 files.
Online, there are 2174 of type: Text available.

Closing browser...

Browser closed.
```

**Pull request merged**